dirstore.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/mendersoftware/log"
+)
+
+type DirStore struct {
+	basepath string
+}
+
+type DirFile struct {
+	io.WriteCloser
+	name     string
+	dirstore *DirStore
+}
+
+func NewDirStore(path string) *DirStore {
+	return &DirStore{
+		basepath: path,
+	}
+}
+
+func (d DirStore) ReadAll(name string) ([]byte, error) {
+	in, err := d.OpenRead(name)
+	if err != nil {
+		return nil, err
+	}
+	defer in.Close()
+	return ioutil.ReadAll(in)
+}
+
+func (d DirStore) WriteAll(name string, data []byte) error {
```

Comment:
Oh, wait. In fact we won't send a request so next time we will increase seq number as expected. Ignore my comment above.